android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,244 @@
+            language = "und"
+        } else if (language == "iw") {
+            language = "he"
+        } else if (language == "in") {
+            language = "id"
+        } else if (language == "ji") {
+            language = "yi"
+        }
+
+        if (!region.matches("\\\\[a-zA-Z]{2}|\\\\[0-9]{3}".toRegex())) {
+            region = ""
+        }
+
+        if (!variant.matches("\\\\[a-zA-Z0-9]{5,8}|\\\\[0-9]\\\\[a-zA-Z0-9]{3}".toRegex())) {
+            variant = ""
+        }
+
+        val bcp47Tag = StringBuilder(language)
+
+        if (region.isNotEmpty()) {
+            bcp47Tag.append(sep).append(region)
+        }
+
+        if (variant.isNotEmpty()) {
+            bcp47Tag.append(sep).append(variant)
+        }
+
+        return bcp47Tag.toString()
+    }
+
+    @SuppressLint("HardwareIds")
+    private fun makeDeviceObj(): JsonObject {
+        return JsonObject().apply {
+            val pref = applicationContext.getSharedPreferences("api", Context.MODE_PRIVATE)
+
+            if (pref.contains("deviceId")) {
+                addProperty("id", pref.getString("deviceId", null))
+            }
+
+            addProperty("language", language())
+            add("platform", JsonObject().apply {
+                addProperty("version", Build.VERSION.RELEASE)
+                addProperty("sdkVersion", Build.VERSION.SDK_INT)
+                addProperty("brand", Build.BRAND)
+                addProperty("model", Build.MODEL)
+            })
+
+            addProperty("timezone", Calendar.getInstance().timeZone.displayName)
+            addProperty("type", "android")
+
+            try {
+                addProperty("version", applicationContext.packageManager.getPackageInfo(applicationContext.packageName, 0).versionName)
+            } catch (e: Exception) {
+                addProperty("version", "unknown")
+            }
+
+            add("screen", JsonObject().apply {
+                val manager = applicationContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+                val display = manager.defaultDisplay
+                val size = Point()
+                display.getSize(size)
+                addProperty("width", size.x)
+                addProperty("height", size.y)
+            })
+        }
+    }
+
+    @SuppressLint("HardwareIds")
+    suspend fun makeRequest(
+        functionName: String,
+        bodyArgs: JsonObject?,
+        timeoutMillis: Long? = null
+    ): InternalResponse = suspendCoroutine { continuation ->
+        try {
+            val body = JsonObject().apply {
+                addProperty("version", 3)
+                addProperty("requestId", callId())
+                addProperty("name", functionName)
+                add("args", bodyArgs ?: JsonObject())
+                add("extra", JsonObject().apply { addProperty("fingerprint", Settings.Secure.getString(applicationContext.contentResolver, Settings.Secure.ANDROID_ID)) })
```

Comment:
fingerprint should go inside devices's platform

Reply:
I'll put the fingerprint inside the deviceInfo as discussed in private.